Problem-3.py

```diff
@@ -0,0 +1,52 @@
+"""
+## Problem3 (https://leetcode.com/problems/search-a-2d-matrix-ii/)
+Write an efficient algorithm that searches for a value in an m x n matrix. This matrix has the following properties:
+
+Integers in each row are sorted in ascending from left to right.
+Integers in each column are sorted in ascending from top to bottom.
+Example:
+
+Consider the following matrix:
+
+[
+  [1,   4,  7, 11, 15],
+  [2,   5,  8, 12, 19],
+  [3,   6,  9, 16, 22],
+  [10, 13, 14, 17, 24],
+  [18, 21, 23, 26, 30]
+]
+Given target = 5, return true.
+
+Given target = 20, return false.
+
+TIme - 0(m+n)
+Space - 0(m+n)
```

Comment:
What space are you using?

Reply:
OOPS, sorry its constant space, I am not using any extra space.